station/app/config.py

```diff
@@ -519,6 +535,108 @@ def _setup_minio_connection(self):
             else:
                 logger.warning(f"No minio config specified in config or env vars, ignoring in development mode")
 
+    def _setup_airflow(self):
+
+        logger.info(f"Setting up airflow connection and airflow-connection to station database...")
+
+        # get the environment variables for airflow
+        env_airflow_host, env_airflow_port, env_airflow_user, env_airflow_secret = self._get_internal_service_env_vars(
+            host=StationEnvironmentVariables.AIRFLOW_API_URL,
+            port=StationEnvironmentVariables.AIRFLOW_PORT,
+            user=StationEnvironmentVariables.AIRFLOW_USER,
+            secret=StationEnvironmentVariables.AIRFLOW_PW)
+
+
+        # get airflow from config file or construct dummy
+        _airflow_config = False
+        airflow_config = self.config.airflow
+        if airflow_config:
+            _airflow_config = True
+            logger.debug("Airflow config found, checking for env vars.")
+        else:
+            airflow_config = AirflowSettings.construct()
+
+        # override airflow config if config and env vars are present and validate afterwards
+        if _airflow_config and (env_airflow_host or env_airflow_port or env_airflow_user or env_airflow_secret):
+            logger.debug(f"Overriding airflow config with env var specifications.")
+            if env_airflow_host:
+                airflow_config.host = env_airflow_host
+            if env_airflow_port:
+                airflow_config.port = env_airflow_port
+            if env_airflow_user:
+                airflow_config.user = env_airflow_user
+            if env_airflow_secret:
+                airflow_config.password = env_airflow_secret
+            _airflow_config = True
+        # no airflow config found
+        elif not _airflow_config and not (env_airflow_host and env_airflow_user and env_airflow_secret):
+            _airflow_config = False
+
+        # no config but environment variables are found
+        elif not _airflow_config and (env_airflow_host and env_airflow_user and env_airflow_secret):
+            logger.debug(f"{Emojis.INFO}No airflow config found, creating new one from env vars.")
+            airflow_config.host = env_airflow_host
+            airflow_config.user = env_airflow_user
+            airflow_config.password = env_airflow_secret
+            if env_airflow_port:
+                airflow_config.port = env_airflow_port
+            _airflow_config = True
```

Comment:
That's a bit confusing still, primarily api url != host. host would be airflow or localhost